src/sentry/api/event_search.py

```diff
@@ -1417,7 +1417,11 @@ def get_json_meta_type(field_alias, snuba_type, function=None):
     return snuba_json
 
 
-FUNCTION_PATTERN = re.compile(r"^(?P<function>[^\(]+)\((?P<columns>[^\)]*)\)$")
+FUNCTION_PATTERN = re.compile(
+    # The alias for alias here is to match any word, but exclude anything that is only digits
+    # eg. 123 doesn't match, but test_123 will match
+    r"^(?P<function>[^\(]+)\((?P<columns>[^\)]*)\)( (as|AS) (?P<alias>(\w+)?(?!\d+)\w+))?$"
```

Comment:
```suggestion
    r"^(?P<function>[^\(]+)\((?P<columns>[^\)]*)\)( (as|AS) (?P<alias>[a-zA-Z]\w*))?$"
```
Is a bit simpler and is effectively the same.

Reply:
heh @Zylphrex had the same suggestion, but this wouldn't work if the alias starts with a digit (eg. `123_test`)